src/components/ProxyGroup/ProxyEditItem/ProxyEditItem.js

```diff
@@ -136,43 +139,31 @@ class ProxyEditItem extends React.Component {
       name,
       record,
       onDelete,
+      intl: { formatMessage }
     } = this.props;
 
-    const relationStatusValues = [
-      <FormattedMessage id="ui-users.active" />,
-      <FormattedMessage id="ui-users.inactive" />,
-    ];
-    const proxySponsorValues = [
-      <FormattedMessage id="ui-users.sponsor" />,
-      <FormattedMessage id="ui-users.proxy" />,
-    ];
-    const yesNoValues = [
-      <FormattedMessage id="ui-users.yes" />,
-      <FormattedMessage id="ui-users.no" />,
-    ];
-
-    const relationStatusOptions = relationStatusValues.map(val => ({
-      label: val,
-      value: val,
-      selected: record.proxy && record.proxy.status === val
+    const relationStatusOptions = ['active', 'inactive'].map(option => ({
```

Comment:
@zburke, it is not necessary to inject `intl` for the options of `Select` and we can use `<FormattedMessage>` in the following way:

```
const requestForSponsorOptions = ['yes', 'no'].map(option => (
    <FormattedMessage id={`ui-users.${option}`} >
    {(optionTranslated) => <option value={option}> {optionTranslated}</option>}
    </FormattedMessage>
));
```
and then in the Select we do not need the dataOptions prop:

```
<Field
  disabled={this.state.statusDisabled}
  label={<FormattedMessage id="ui-users.proxy.relationshipStatus" />}
  name={`${name}.proxy.status`}
  component={Select}
  fullWidth
  defaultValue={record.proxy && record.proxy.status}
>
  {requestForSponsorOptions}
</Field>
```

Reply:
Good point; thanks for the example. I'll refactor.